animal_soup/extensions/sequence_extensions.py

```diff
@@ -0,0 +1,418 @@
+                            sequence_length=DEFAULT_DATA_PARAMS["sequence_length"]
+                            )
+
+        dataset_metadata = datasets.dataset_info
+
+        # create TGMJ model, no pre-trained weights needed
+        num_classes = len(BEHAVIOR_CLASSES) + 1 # account for background
+
+        # reload weights from file, want to use pretrained weights
+        model, model_in = _load_pretrained_sequence_model(
+            weight_path=model_in,
+            exp_type=exp_type,
+            num_classes=num_classes,
+            num_neg=datasets.num_neg,
+            num_pos=datasets.num_pos
+        )
+
+        # lightning module
+        lightning_module = SequenceLightningModule(
+            sequence_model=model,
+            gpu_id=gpu_id,
+            datasets=datasets,
+            initial_lr=initial_lr,
+            batch_size=batch_size,
+            model_in=model_in
+        )
+
+        # trainer
+        trainer = get_sequence_trainer(
+            gpu_id=gpu_id,
+            model_out=model_out,
+            stop_method=stop_method
+        )
+
+        model_params = {
+            "Sequence Model": 'TGMJ',
+            "Parameters": {
+                "initial_learning_rate": initial_lr,
+                "batch_size": batch_size,
+                "stop_method": stop_method,
+            },
+            "Weight Path": model_in,
+            "Output Path": model_out,
+        }
+
+        print("Starting training")
+        print("Model Parameters:")
+        pprint.pprint(model_params)
+        print("Data Info: ")
+        pprint.pprint(dataset_metadata)
+
+        if "model_params" not in self._df.columns:
+            self._df.insert(
+                loc=len(self._df.columns) - 1,
+                column="model_params",
+                value=[dict() for i in range(len(self._df.index))],
+            )
+
+        # add sequence model params to df
+        for ix in range(len(self._df.index)):
+            self._df.loc[ix]["model_params"].update(
+                {"sequence_train": f"{model_params}"}
+            )
+        # save df
+        self._df.behavior.save_to_disk()
+
+        trainer.fit(lightning_module)
+
+
+@pd.api.extensions.register_series_accessor("sequence")
+class SequenceModelSeriesExtensions:
+    """Pandas series extensions for inference of the sequence model."""
+
+    def __init__(self, s: pd.Series):
+        self._series = s
+
+    def infer(
+            self,
+            model_in: Union[str, Path] = None
+    ):
+        """
+
+        Parameters
+        ----------
+        model_in: str or Path, default None
+            If you want to retrain the model using different model weights than the default. User can
+            provide a location to a different model checkpoint. For example, if you had retrained the sequence model
+            previously and wanted to use those weights for inference instead.
```

Comment:
looks like you copy pasted from train? 

```suggestion
            If you want to use your own model instead of the default you can provide a location to a different model checkpoint. For example, if you retrained the sequence model for a new behavioral task or setup
             and want to use those weights for inference instead of the default models.
```